rs/bitcoin/kyt/src/state.rs

```diff
@@ -24,9 +24,25 @@ pub enum HttpGetTxError {
     Rejected {
         code: RejectionCode,
         message: String,
+        provider: Provider,
     },
 }
 
+impl HttpGetTxError {
+    pub fn is_retriable(&self) -> bool {
+        match self {
+            HttpGetTxError::Rejected { code, .. } => matches!(code, RejectionCode::SysTransient),
+            _ => false,
```

Comment:
In case a provider is malicious, we could encounter a `TxidMismatch` error. Shouldn't we then try another provider in that case?

Reply:
Since we don't trust any provider, the more general question is: Is there any scenario where we do not ask other providers if an error is encountered?
Since the error is tied to the provider's response, I can't think of a scenario myself...